### PR TITLE
Get schema from documentType while YamlMime is YamlDocument

### PR DESCRIPTION
**Pull request by live1206**
https://ceapex.visualstudio.com/Engineering/_workitems/edit/87522

Get `schema` from `documentType` while `YamlMime` is `YamlDocument`

### CODE REVIEW

File: src/docfx/build/page/BuildPage.cs [hunk 1]
```diff
@@ -159,10 +159,8 @@ private static (List<Error> errors, Schema schema, OutputModel model)
         private static async Task<(List<Error> errors, Schema schema, OutputModel model)>
             LoadSchemaDocument(Context context, List<Error> errors, JToken token, Document file, Action<Document> buildChild)
         {
-            // TODO: for backward compatibility, when #YamlMime:YamlDocument, documentType is used to determine schema.
-            //       when everything is moved to SDP, we can refactor the mime check to Document.TryCreate
             var obj = token as JObject;
-            var schema = file.Schema ?? Schema.GetSchema(obj?.Value<string>("documentType"));
+            var schema = file.Schema;
```
Comment (OsmondJiang) [inline src/docfx/build/page/BuildPage.cs]: directly use file.Schema?

File: src/docfx/lib/json/YamlUtility.cs [hunk 2]
```diff
@@ -22,7 +22,12 @@ internal static class YamlUtility
 
         public static string ReadMime(TextReader reader)
         {
-            return ReadMime(reader.ReadLine());
+            var mime = ReadMime(reader.ReadLine());
+            if (string.Compare(mime, "YamlDocument", StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return ReadDocumentType(reader.ReadLine());
```
Comment (yufeih) [inline src/docfx/lib/json/YamlUtility.cs]: Im not sure if documenttype ia always the first property

Comment (live1206) [reply src/docfx/lib/json/YamlUtility.cs]: right

File: src/docfx/lib/json/YamlUtility.cs [hunk 3]
```diff
@@ -22,7 +22,12 @@ internal static class YamlUtility
 
         public static string ReadMime(TextReader reader)
```
Comment (superyyrrzz) [inline src/docfx/lib/json/YamlUtility.cs]: `ReadMime` => `ReadSchema`?

Comment (live1206) [reply src/docfx/lib/json/YamlUtility.cs]: I will leave it as it is here, we have another method `ReadSchema` reading `$schema` from JSON

**Pull request merged**